model/labels/matcher.go

```diff
@@ -78,7 +79,30 @@ func MustNewMatcher(mt MatchType, name, val string) *Matcher {
 }
 
 func (m *Matcher) String() string {
-	return fmt.Sprintf("%s%s%q", m.Name, m.Type, m.Value)
+	const quote = 1
+	const matcher = 2
```

Comment:
Maybe call these `lenQuote` and `lenMatcher` or something? I first thought this might be keys to different types of characters.

Reply:
It's a valid point, I'll do it in a follow up PR. Right now I removed pre-growing at all, as I don't have a benchmark for it and there's now a 1KB preallocated stack for building the matcher which covers all the benchmarks.

Let's merge this as is, and I can add a 4KB long matcher to the benchmarks in a follow up PR, and see the impact of growing.